webapp/components/suggestion/switch_channel_provider.jsx

```diff
@@ -57,83 +67,179 @@ class SwitchChannelSuggestion extends Suggestion {
+                } else {
+                    wrappedChannel.type = Constants.MENTION_MORE_CHANNELS;
+                    if (skipNotInChannel || !newChannel.display_name.startsWith(channelPrefix)) {
+                        continue;
+                    }
+                }
 
-                    const channelNames = channels.
-                        sort(sortChannelsByDisplayName).
-                        map((channel) => channel.name);
-
-                    AppDispatcher.handleServerAction({
-                        type: ActionTypes.SUGGESTION_RECEIVED_SUGGESTIONS,
-                        id: suggestionId,
-                        matchedPretext: channelPrefix,
-                        terms: channelNames,
-                        items: channels,
-                        component: SwitchChannelSuggestion
-                    });
-
-                    AppDispatcher.handleServerAction({
-                        type: ActionTypes.RECEIVED_PROFILES,
-                        profiles: userMap
-                    });
+                channels.push(wrappedChannel);
+            }
+        }
+
+        for (let i = 0; i < users.length; i++) {
+            const user = users[i];
+            const isDMVisible = getBool(getState(), Preferences.CATEGORY_DIRECT_CHANNEL_SHOW, user.id, false);
+            let displayName = `@${user.username} `;
+
+            if (user.id === currentId) {
+                continue;
+            }
+
+            if ((user.first_name || user.last_name) && user.nickname) {
+                displayName += `- ${Utils.getFullName(user)} (${user.nickname})`;
+            } else if (user.nickname) {
+                displayName += `- (${user.nickname})`;
+            } else if (user.first_name || user.last_name) {
+                displayName += `- ${Utils.getFullName(user)}`;
+            }
+
+            const wrappedChannel = {
+                channel: {
+                    display_name: displayName,
+                    name: user.username,
+                    id: user.id,
+                    update_at: user.update_at,
+                    type: Constants.DM_CHANNEL
+                },
+                name: user.username
+            };
+
+            if (isDMVisible) {
+                wrappedChannel.type = Constants.MENTION_CHANNELS;
+            } else {
+                wrappedChannel.type = Constants.MENTION_MORE_CHANNELS;
+                if (skipNotInChannel) {
+                    continue;
                 }
-            );
+            }
+
+            channels.push(wrappedChannel);
         }
+
+        const channelNames = channels.
+            sort(quickSwitchSorter).
+            map((wrappedChannel) => wrappedChannel.channel.name);
+
+        if (skipNotInChannel) {
+            channels.push({
+                type: Constants.MENTION_MORE_CHANNELS,
+                loading: true
+            });
+        }
+
+        setTimeout(() => {
```

Comment:
Yeah should be fixed when we move the suggestion store to be in the view state stuff coming with my post list PR